src/OpenTelemetry.Exporter.Prometheus/MeterProviderBuilderExtensions.cs

```diff
@@ -39,13 +39,11 @@ public static MeterProviderBuilder AddPrometheusExporter(this MeterProviderBuild
             configure?.Invoke(options);
 
             var exporter = new PrometheusExporter(options);
-
-            var metricReader = new BaseExportingMetricReader(exporter);
-            exporter.CollectMetric = metricReader.Collect;
+            var reader = new BaseExportingMetricReader(exporter);
 
             var metricsHttpServer = new PrometheusExporterMetricsHttpServer(exporter);
             metricsHttpServer.Start();
```

Comment:
@alanwest I noticed this while changing the code.

I think we _might_ run into some race condition - if a scraper happens to hit the HTTP server before we could add the reader, what would happen (I guess we will hit exception, which turns into HTTP 500)? I haven't looked into the HTTP server logic.

I think it _might_ be OKAY. A better version might be - we only start the HTTP server once the exporter/reader are fully ready and both are hooked up to the provider.

Reply:
if the reader is not ready, we could modify the http server to still return 200, but with empty metric. 
try
{
this.exporter.Collect(Timeout.Infinite)
}
catch(Exporter/Reader not readyexception)
{
 exporter.BatchMetrics = default.
}

Or we can wait for some signal before starting HTTP Server. 
Will track this as a separate follow up once this PR is merged.